moveos/moveos-stdlib/moveos-stdlib/sources/object.move

```diff
@@ -125,19 +132,27 @@ module moveos_std::object {
     /// The shared object also can be removed from the object storage.
     public fun to_shared<T: key>(self: Object<T>) {
         let obj_entity = borrow_mut_from_global<T>(self.id);
-        // TODO set the flag
-        transfer_to_system(obj_entity); 
+        to_shared_internal(obj_entity);
         to_permanent(self);
     }
 
+    fun to_shared_internal<T: key>(self: &mut ObjectEntity<T>) {
+        self.flag = self.flag | SHARED_OBJECT_FLAG_MASK;
```

Comment:
Should be reset directly to `SHARED_OBJECT_FLAG_MASK` here ? 
Shared objects can be set to a frozen state? If yes,  there will be problems when setting frozen objects to be shared.
Or if the flag is expanded later, there may appear a bug here?

Reply:
Shared Object and frozen Object can not be at the same time, but I added a test to ensure the flag can be set at the same time.